src/types/internal/schema.ts

```diff
@@ -466,6 +469,6 @@ export class Schema {
     return this.names.slice().sort().join(' ') + '/' +
       Object.keys(this.fields).sort().map(field =>
         `${field}:${this.normalizeTypeForHash(this.fields[field])}`
-      ).join('');
+      ).join('') + '/';
```

Comment:
I wonder if we are also better off having a separator between fields eg
).join('-') + '/';

Reply:
there is a separator, of sorts, for fields that don't wrap (the vertical pipe). Perhaps it's better to remove it from the per-field export and do it systematically?

I'm also considering just retiring this method entirely and modifying the toString() method to take an optional sort parameter. Seems crazy that we have a guaranteed-unique representation for parsing that we don't use.

Reply:
I've left as-is for now, (updating the hashes in all our tests is painful and effectively there is already a separator). I've filed the idea of moving to toString() as an issue (b/168267507).